packages/relay/src/lib/eth.ts

```diff
@@ -1932,16 +1932,29 @@ export class EthImpl implements Eth {
   }
 
   private async getAccountLatestEthereumNonce(address: string, requestId?: string) {
```

Comment:
The goal here was to wait for the HIP 729 implementation and then replace `isValidContract`.
Since the Mirror Node also manages accounts and contracts on the `api/v1/accounts` endpoint this whole method may just get simplified to the single `getAccount` call.

I think rather another consideration in the mean time is to
1. update mirrorNodeClient.getContract to populate the same cache that isValidContract gets.
2. [Stretch] Add an isValidAccount() and update mirrorNodeClient.getAccount to populate the same cache that isValidAccount gets. This was the intention but wasn't needed when improving `eth_getTransactionCount()`

Both these optimization should help reduce the number of calls to resolveEntityType